sdk/storage/azure-storage-file/tests/test_file_client.py

```diff
@@ -244,6 +244,78 @@ def test_create_service_with_connection_string_succeeds_if_secondary_with_primar
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/?'))
+
+        service = ShareClient(account_url=custom_account_url, share_name="foo", snapshot="snap")
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.share_name, "foo")
+        self.assertEqual(service.snapshot, "snap")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/foo?sharesnapshot=snap&'))
+
+        service = DirectoryClient(account_url=custom_account_url, share_name='foo', directory_path="bar/baz", snapshot="snap")
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.share_name, "foo")
+        self.assertEqual(service.directory_path, "bar/baz")
+        self.assertEqual(service.snapshot, "snap")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        # TODO: Should directory path be URL-encoded here?
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/foo/bar%2Fbaz?sharesnapshot=snap&'))
+
+        service = DirectoryClient(account_url=custom_account_url, share_name='foo', directory_path="")
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.share_name, "foo")
+        self.assertEqual(service.directory_path, "")
+        self.assertEqual(service.snapshot, None)
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/foo?'))
+
+        service = FileClient(account_url=custom_account_url, share_name="foo", file_path="bar/baz/file", snapshot="snap")
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.share_name, "foo")
+        self.assertEqual(service.directory_path, "bar/baz")
+        self.assertEqual(service.file_path, ["bar", "baz", "file"])
+        self.assertEqual(service.file_name, "file")
+        self.assertEqual(service.snapshot, "snap")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        # TODO: Should file path be URL-encoded here?
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/foo/bar/baz/file?sharesnapshot=snap&'))
```

Comment:
Hi @xiafu-msft 
As you can see here, the `FileClient` is url-encoding _only the components within the path_. So "bar/baz/file" to "bar/baz/file". This behaviour does not seem correct? (I have not yet altered this - it's the current package behaviour)

Reply:
It should also be noted that the live tests pass regardless of whether we use URL-encoding here.... which makes me wonder if the SDK isn't behaving the way we think it is, or whether the service simply supports both.

Reply:
Actually it doesn't matter because file service will unencode the url. ^_^

Reply:
Sweet, okay. No changes needed then :)